### PR TITLE
Upgraded Point encoder to support many more types

### PR DESCRIPTION
**Pull request by bminer**
As discussed previously, this PR will allow Points to be encoded / decoded to / from many more Go types, including arrays, maps, and "flat" structs.

### CODE REVIEW

**Comment by cbrake** [comment]
looking good -- thanks for adding tests. Do you plan to also add decode?

File: client/manager.go [hunk 1]
```diff
@@ -40,8 +40,7 @@ type Manager[T any] struct {
 func NewManager[T any](nc *nats.Conn,
 	construct func(nc *nats.Conn, config T) Client) *Manager[T] {
 	var x T
-	nodeType := reflect.TypeOf(x).Name()
-	nodeType = strings.ToLower(nodeType[0:1]) + nodeType[1:]
+	nodeType := data.ToCamelCase(reflect.TypeOf(x).Name())
```
Comment (bminer) [inline client/manager.go:43]: Is the `data` package the best place for this?

File: data/decode.go [hunk 2]
```diff
@@ -13,6 +13,100 @@ type NodeEdgeChildren struct {
 	Children []NodeEdgeChildren
 }
 
+type GroupedPoints struct {
```
Comment (bminer) [inline data/decode.go:16]: Comment probably needed

File: data/decode.go [hunk 3]
```diff
@@ -13,6 +13,100 @@ type NodeEdgeChildren struct {
...
+	Keyed    bool
+	IndexMax int
+	Points   []Point
+}
+
+var reflectValueT = reflect.TypeOf(reflect.Value{})
+
+func setValueFromPointGroup(g GroupedPoints, v reflect.Value) error {
```
Comment (bminer) [inline data/decode.go:24]: Maybe make this a method of the GroupedPoints struct?

File: data/decode.go [hunk 4]
```diff
@@ -31,71 +125,117 @@ type NodeEdgeChildren struct {
...
+			if err != nil {
+				log.Printf("decode error for type %v: %v", pt, err)
+			}
 		} else if et := sf.Tag.Get("edgepoint"); et != "" {
-			edgeValues[et] = vOut.Field(i)
+			g := edgePointGroups[et]
+			// Write points into struct field
+			err := setValueFromPointGroup(g, vOut.Field(i))
+			if err != nil {
+				log.Printf("decode error for type %v: %v", et, err)
```
Comment (bminer) [inline data/decode.go:193]: Should these return an error instead of printing to console?

File: data/merge.go [hunk 5]
```diff
@@ -8,17 +8,43 @@ import (
...
-	case reflect.String:
-		v.SetString(p.Text)
-	case reflect.Int:
-		v.SetInt(int64(p.Value))
-	case reflect.Float64, reflect.Float32:
-		v.SetFloat(p.Value)
+	t := v.Type()
+	k := t.Kind()
+	if !v.CanSet() {
+		log.Println("setVal failed, cannot set value")
```
Comment (bminer) [inline data/merge.go:14]: Should these return an error instead of printing to console?

**Review by bminer — COMMENTED** [review]
@cbrake - Please let me know your thoughts! Thanks!

File: client/manager.go (hunk 1 above)
Comment (cbrake) [reply client/manager.go:43]: that is fine with me -- a lot of people claim you should not have a data package, but I'm not there yet ...

File: data/decode.go (hunk 2 above)
Comment (cbrake) [reply data/decode.go:16]: :+1:

File: data/decode.go (hunk 3 above)
Comment (cbrake) [reply data/decode.go:24]: yes, I would probably lean that way.

File: data/decode.go (hunk 4 above)
Comment (cbrake) [reply data/decode.go:193]: if the function returns an error, then yes. 

I also learned recently the fmt.Errorf should use %w for err, as then it wraps the error. You probably know this.

File: data/merge.go (hunk 5 above)
Comment (cbrake) [reply data/merge.go:14]: generally yes. The only reason I can think of to log an error and not return is there is some value in a partial merge.

**Comment by cbrake** [comment]
`siot_test` fails with a number of build failures, so we'll need to fix those up before merging. This also runs the linter to check if things are clean.

File: data/merge.go (hunk 5 above)
Comment (bminer) [reply data/merge.go:14]: So going forward, `setVal` returns an error. The system will continue to merge points even if an error occurs, and the Decode function will return the first error that occurred.

**Comment by bminer** [comment]
Ready to review / merge. Thanks for your feedback!

**Comment by cbrake** [comment]
thanks @bminer  -- nice work!

**Pull request merged**